static/src/javascripts/projects/common/modules/experiments/tests/recommended-for-you.js

```diff
@@ -112,45 +140,71 @@ define([
-                'imageUrl': imageUrlFromItem(item.item),
-                'title': item.item.title,
-                'standFirst': item.item.standFirst
-            };
-        }
-
         function setupComponentAttentionTracking(trackingCode) {
             require(['ophan/ng'], function (ophan) {
                 ophan.trackComponentAttention(trackingCode, $recommendedForYouSection[0]);
             });
         }
 
         function insertSection(items) {
+            var $oldSection = $recommendedForYouSection;
+
             $recommendedForYouSection = $.create(template(recommendedForYouTemplate, {
                 profileIcon: svg(profileIcon, ['rounded-icon', 'rfy-profile-icon', 'control__icon-wrapper']),
-                guardianLogo: svg(guardianLogo),
                 items: items
             }));
 
+            if ($oldSection != null) {
+                return fastdom.write(function() {
+                    $oldSection.replaceWith($recommendedForYouSection);
+                    setupComponentAttentionTracking('recommended-for-you_user-history');
+                });
+            } else {
+                return fastdom.write(function() {
+                    $recommendedForYouSection.insertBefore($opinionSection);
+                    setupComponentAttentionTracking('recommended-for-you_user-history');
+                });
+            }
+        }
+
+        function hasGivenFeedback() {
+            return storage.local.get(cachedRecommendationsEnabledKey) != null;
+        }
+
+        function shouldShowRecommendations() {
+            return !!storage.local.get(cachedRecommendationsEnabledKey);
+        }
+
+        function registerFeedback(showRecommendations) {
+            storage.local.set(cachedRecommendationsEnabledKey, showRecommendations);
+        }
+
+        function insertOnBoardingSection() {
+            $recommendedForYouSection = $.create(template(recommendedForYouOptInTemplate, {
+                profileIcon: svg(profileIcon, ['rounded-icon', 'rfy-profile-icon', 'control__icon-wrapper']),
+                rightArrowIcon: svg(rightArrowIcon, ['i-right']),
+                guardianLogo: svg(guardianLogo)
+            }));
+
             return fastdom.write(function() {
                 $recommendedForYouSection.insertBefore($opinionSection);
+                $('.js-feedback-button-yes', $recommendedForYouSection[0]).each(function(el) {
+                    bean.on(el, 'click', function () {
+                        registerFeedback(true);
+                        $('.js-feedback', $recommendedForYouSection[0]).html(
+                            '<p>Your recommendations will be ready soon.</p>'
+                        );
+                        populateRecommendationsContainer();
+                    });
+                });
+
+                $('.js-feedback-button-no', $recommendedForYouSection[0]).each(function(el) {
+                    bean.on(el, 'click', function () {
```

Comment:
If we assume there should never be more than one `js-feedback-button-no`, we could remove the `.each(`:

```
bean.on($('.js-feedback-button-no', $recommendedForYouSection)[0], 'click', function () {
    registerFeedback(false);
    $recommendedForYouSection.remove();
});
```

(and the same for `.js-feedback-button-yes`)